components/reset_status_modal/reset_status_modal.jsx

```diff
@@ -58,32 +71,77 @@ export default class ResetStatusModal extends React.PureComponent {
         );
     }
 
-    onConfirm = (checked) => {
+    componentWillUnmount() {
+        ModalStore.removeModalListener(ActionTypes.TOGGLE_RESET_STATUS_MODAL, this.toggleModal);
+    }
+
+    toggleModal = (show, args) => {
+        const {currentUserStatus} = this.props;
+
+        this.setState({
+            show,
+            newStatus: args.newStatus || 'online',
+            currentUserStatus: {
+                ...this.state.currentUserStatus,
+                status: currentUserStatus,
+            },
+        });
+    };
+
+    hideModal = () => {
         this.setState({show: false});
+    };
+
+    onConfirm = (checked) => {
+        this.hideModal();
 
         const newStatus = {...this.state.currentUserStatus};
-        newStatus.status = 'online';
+        newStatus.status = this.state.newStatus;
         this.props.actions.setStatus(newStatus);
 
         if (checked) {
             const pref = {category: Preferences.CATEGORY_AUTO_RESET_MANUAL_STATUS, user_id: newStatus.user_id, name: newStatus.user_id, value: 'true'};
             this.props.actions.savePreferences(pref.user_id, [pref]);
         }
-    }
+    };
 
     onCancel = (checked) => {
-        this.setState({show: false});
+        this.hideModal();
 
         if (checked) {
             const status = {...this.state.currentUserStatus};
             const pref = {category: Preferences.CATEGORY_AUTO_RESET_MANUAL_STATUS, user_id: status.user_id, name: status.user_id, value: 'false'};
             this.props.actions.savePreferences(pref.user_id, [pref]);
         }
-    }
+    };
+
+    renderModalMessage = () => {
```

Comment:
Please add test case for this - https://github.com/mattermost/mattermost-webapp/blob/master/tests/components/reset_status_modal.test.jsx